### DISCUSSION TITLE
Add SQL DataSource

### DISCUSSION THREAD

**Discussion by zackkatz** [UseDataKit/SDK#21] [Ideas]
This source would be passed a SQL query and would display the results in a DataView.

The query will generate the starting names and order of columns in the response, then, by adding fields to the table, you can override the settings and define a different data type.

It would make sense to me that fields were not required for the view; that all you would need would be to pass the query datasource to the view.

**Comment by doekenorg** [comment]
I like the idea of providing the (base) query, as you can create highly optimized queries including joins on other tables. 

But when it comes to filtering and sorting, this needs to happen through the `Sort` and `Filters` objects. Otherwise we can't provide the proper API for DataViews.

As for the fields, the data source knows nothing about the fields. And it can only provide the data, and not the field types you want to use. It can provide the available columns on the `get_fields()` method.

This also makes it compatible for future use with the a UI.
